torch/csrc/jit/codegen/cuda/codegen.cpp

```diff
@@ -1415,6 +1416,76 @@ class CudaKernelGenerator : private OptOutConstDispatch {
     indent() << kTab << func_args << ");\n";
   }
 
+  void handle(const kir::GroupedGridReduction* grouped_grop) final {
+    const auto out = ir_utils::getTvOutput(grouped_grop);
+    const auto domain = out->domain();
+    TORCH_INTERNAL_ASSERT(domain->hasGridReduction());
+
+    TORCH_INTERNAL_ASSERT(
+        grouped_grop->sync_buffer()->buffer()->isA<TensorView>());
+    const auto sync_buffer =
+        grouped_grop->sync_buffer()->buffer()->as<TensorView>();
+
+    TORCH_INTERNAL_ASSERT(
+        grouped_grop->numReductions() == 2,
+        "Only grouping of 2 reductions is supported. ",
+        grouped_grop->toString());
+
+    // TODO: enable this
+    TORCH_INTERNAL_ASSERT(!grouped_grop->isFused());
+
+    const std::string flags_str = generateGridReduceTemplateFlags2(
+        grouped_grop, grouped_grop->threadPredicate());
+
+    const bool persistent_sync =
+        kernel_->summary().has_cooperative_grid_reduction;
```

Comment:
I wonder if at some point we would want a mix of persistent grid syncs and non-persistent grid syncs in the same kernel.